atomic_reactor/plugins/pre_pin_operator_digest.py

```diff
@@ -171,24 +171,35 @@ def _get_replacement_pullspecs(self, pullspecs):
             if not replacer.registry_is_allowed(p):
                 raise RuntimeError("Registry not allowed: {} (in {})".format(p.registry, p))
 
+        pin_digest = self.user_config.get("enable_digest_pinning", True)
+        replace_repo = self.user_config.get("enable_repo_replacements", True)
+
+        if not pin_digest:
+            self.log.warning("User disabled digest pinning")
+        if not replace_repo:
+            self.log.warning("User disabled repo replacements")
+
         replacements = {}
 
         for original in pullspecs:
             self.log.info("Computing replacement for %s", original)
+            replaced = original
 
-            self.log.debug("Making sure tag is manifest list digest")
-            pinned = replacer.pin_digest(original)
+            if pin_digest:
+                self.log.debug("Making sure tag is manifest list digest")
+                replaced = replacer.pin_digest(original)
 
-            self.log.debug("Replacing namespace/repo")
-            repo_replaced = replacer.replace_repo(pinned)
+            if replace_repo:
+                self.log.debug("Replacing namespace/repo")
+                replaced = replacer.replace_repo(replaced)
 
             self.log.debug("Replacing registry")
-            registry_replaced = replacer.replace_registry(repo_replaced)
+            replaced = replacer.replace_registry(replaced)
```

Comment:
I'm thinking if we should also add option for disabling registry replacements too

Reply:
Would make sense. If all the replacements are disabled, should the plugin still check allowed_registries or just be skipped completely?

Reply:
it makes sense to skip it then, yup